content/en/blog/2022/opamp/index.md

```diff
@@ -0,0 +1,253 @@
+}
+```
+
+Next, add a basic file named “effective.yaml” configuration file. Place it in
+the same folder as our main.go file with this configuration:
+
+```yaml
+instrument: false
+```
+
+Let’s add a basic configuration handler to our server:
+
+```go
+package main
+
+import (
+   "fmt"
+   "gopkg.in/yaml.v3"
+   "io/ioutil"
+   "log"
+   "net/http"
+   "path/filepath"
+)
+
+type configurations struct {
+   Instrument bool
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+   fmt.Fprintf(w, "Hi! This action could create a trace!")
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", handler)
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Next, let's wrap our handler with instrumentation and condition it with our
+configuration file. Something like this:
+
+```go
+package main
+
+import (
+   "context"
+   "fmt"
+   "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+   "go.opentelemetry.io/otel"
+   "go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+   sdktrace "go.opentelemetry.io/otel/sdk/trace"
+   "go.opentelemetry.io/otel/trace"
+   "gopkg.in/yaml.v3"
+   "io/ioutil"
+   "log"
+   "net/http"
+   "os"
+   "path/filepath"
+)
+
+type configurations struct {
+   Instrument bool
+}
+
+var tracer trace.Tracer
+
+func newConsoleExporter() (sdktrace.SpanExporter, error) {
+   return stdouttrace.New(
+      stdouttrace.WithWriter(os.Stdout),
+      stdouttrace.WithPrettyPrint(),
+   )
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+   fmt.Fprintf(w, "Hi! This action could create a trace!")
+}
+
+func setHandler(handler http.Handler, config configurations) http.Handler {
+   if config.Instrument {
+      return otelhttp.NewHandler(handler, "instrumentation activated by OpAMP")
+   }
+   return http.HandlerFunc(httpHandler)
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   exp, _ := newConsoleExporter()
+   tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exp))
+   defer func() { _ = tp.Shutdown(context.Background()) }()
+
+   otel.SetTracerProvider(tp)
+
+   tracer = tp.Tracer("ControlledOpAMPAgentDemo")
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", setHandler(handler, config))
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Build it with `go build .` Now if we run it and open [http://localhost:8080/](http://localhost:8080/) on
+a browser, nothing special will be shown. It’s time to add some opamp. Git clone
+[opamp-go](https://github.com/open-telemetry/opamp-go) and run the server with:
```

Comment:
```suggestion
Build and run this app:

```shell
go build .
<insert run command here>
\``` <-- remove the slash here after merge this suggestion

Open a browser and visit <http://localhost:8080>. Nothing special will be shown.

It’s time to add some opamp. Git clone
[opamp-go](https://github.com/open-telemetry/opamp-go) and run the server with:
```

Reply:
Thanks @chalin, I've merged this change and removed the accidentally committed img file